srcts/src/utils/index.ts

```diff
@@ -336,23 +338,31 @@ const compareVersion = function (
   else throw `Unknown operator: ${op}`;
 };
 
-function updateLabel(
-  labelTxt: string | undefined,
+async function updateLabel(
+  labelContent: string | { html: string; deps: HtmlDep[] } | undefined,
   labelNode: JQuery<HTMLElement>
-): void {
+): Promise<void> {
   // Only update if label was specified in the update method
-  if (typeof labelTxt === "undefined") return;
+  if (typeof labelContent === "undefined") return;
   if (labelNode.length !== 1) {
     throw new Error("labelNode must be of length 1");
   }
 
+  if (typeof labelContent === "string") {
+    labelContent = {
+      html: labelContent,
+      deps: [],
+    };
+  }
+
   // Should the label be empty?
-  const emptyLabel = Array.isArray(labelTxt) && labelTxt.length === 0;
+  const emptyLabel =
+    Array.isArray(labelContent.html) && labelContent.html.length === 0;
```

Comment:
I think this check might also now need to check for empty string? I noticed that this change will break the update logic here:

```r
library(shiny)

ui <- fluidPage(
  checkboxInput("id", label = "foo")
)

server <- function(input, output, session) {
  updateCheckboxInput(inputId = "id", label = character(0))
}

shinyApp(ui, server)
```

Reply:
Ah good catch. Do we still need `Array.isArray` check? I'm not sure what's for to be honest; could we/can we pass a vector of length greater than 1 to `label`? 

Asking because I wonder if it can't be simplified to just `labelContent.html.length == 0` since `"".length == 0` is `true`